### DISCUSSION TITLE
Over/under training (unet and text_encoder) - how to identify on faces

### DISCUSSION THREAD

**Discussion by SorenTruelsen** [TheLastBen/fast-stable-diffusion#1202] [Show and tell]
Hi Guys
In DreamBooth the two main parameters seems to be unet and text_encoder. Most questions here are answered with:
Train unet/text_encoder with XXX steps and then train more if that is not enough. But how do I know if it's not trained enough in either of the two parameters. 

For unet:
"under training" seems to be "the face features are not strongly expressed"
"over training" seems to be "the face gets extremely distorted and any traits gets over expressed"

For text_encoder I am not so sure, it seems to be something like this:
"under training" seems to be "the face only shows up rarely, but when it does its okay"
"over training" seems to be "the face shows up on all people, even Brad Pitt"

Anyone have more observations?

**Comment by Quark999** [comment]
I think this is a very good question that I also don't know the answer to - but I had the impression that "overtraining" manifested itself in "not being able to style properly anymore". I.e. whatever you prompt for, you almost get the exact input pictures back, and can't change the style.
And by the way, "styling" is another term that I see used a lot but am not sure what it is supposed to mean - the above is my interpretation. Maybe we need a glossary!

**Comment by chewtoys** [comment]
always test images with style, like apply watercolor, or oil painting or artist that has a very specific style, if that style dose not show up, but your real human looking face dose, its over trained.

I messed up a lot of models with over training, more then undertraining.

**Comment by TheLastBen** [comment]
if you overtrain the text_encoder, you won't be able to apply any style to the trained subject, that's why 250-400 steps is always enough.
if you overtrain the unet, you will see the trained subject everywhere but you can still apply style to it.
